### DISCUSSION TITLE
Disappearing "used in" information for power queries, lost connection for target cell ranges.

### DISCUSSION THREAD

**Discussion by BoCsA** [ClosedXML/ClosedXML#2399] [General]
Good Morning!

My issue is that excel files which contains power queries only referring cells from the actual excel document as source and as target
loosing their connection to the target cell range to put output data there,
Because of this when I pull a file through ClosedXml open it from stream, do some manipulation and save in the query properties the "used in" data is lost.
Therefore I'm not able to refresh my queries any more after save.
The queries perfectly work  in case of preview, it perfectly collects the modified data from the source cells and show it as preview.
But I cannot load the data into the original output table cause it lost the "used in" information from connection where to load after save.

Is someone struggled with this? Is it a not supported feature of the framework?

When I use OpenXml open the file and save it that doesn't happen.
I just tried without manipulation just opened the file to a stream create a workbook and save it to a new stream and save it to a file.
Unfortunately ClosedXml loses connection between the output table and the query. So I need to manually delete the tables an redefine them. :(

Is there any solution for that? I need to rewrite my whole solution based on ClosedXml  to OpenXml because of this limitation if I didn't find a solution for that.
I would be very grateful for a hint or a solution.
I checked around the saving options, but didn't found any option which solve my issue.

![image](https://github.com/user-attachments/assets/b5ef7544-687f-4ef2-a588-a72bebb0425e)
![image](https://github.com/user-attachments/assets/bed9867b-3676-421d-975c-1352b0418883)

**Comment by joshlupo** [comment]
we're running into the exact same issue, no solution yet =(
